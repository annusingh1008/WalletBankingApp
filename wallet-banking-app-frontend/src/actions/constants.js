export const authConstants = {
    LOGIN_REQUEST: 'LOGIN_REQUEST',
    LOGIN_SUCCESS: 'LOGIN_SUCCESS',
    LOGIN_FAILURE: 'LOGIN_FAILURE',
    LOGOUT_REQUEST: 'LOGOUT_REQUEST',
    LOGOUT_SUCCESS: 'LOGOUT_SUCCESS',
    LOGOUT_FAILURE: 'LOGOUT_FAILURE'
}

export const userConstants = {
    USER_REGISTER_REQUEST: 'USER_REGISTER_REQUEST',
    USER_REGISTER_SUCCESS: 'USER_REGISTER_SUCCESS',
    USER_REGISTER_FAILURE: 'USER_REGISTER_FAILURE',
}

export const amountTransferConstants = {
    AMOUNT_TRANSFER_REQUEST: 'AMOUNT_TRANSFER_REQUEST',
    AMOUNT_TRANSFER_SUCCESS: 'AMOUNT_TRANSFER_SUCCESS',
    AMOUNT_TRANSFER_FAILURE: 'AMOUNT_TRANSFER_FAILURE'
}

export const rechargeConstants = {
    RECHARGE_REQUEST: 'RECHARGE_REQUEST',
    RECHARGE_SUCCESS: 'RECHARGE_SUCCESS',
    RECHARGE_FAILURE: 'RECHARGE_FAILURE',
}

export const getUserDetailsConstants = {
    GET_USER_DETAILS_REQUEST: 'GET_USER_DETAILS_REQUEST',
    GET_USER_DETAILS_SUCCESS: 'GET_USER_DETAILS_SUCCESS',
    GET_USER_DETAILS_FAILURE: 'GET_USER_DETAILS_FAILURE'
}

export const transactionsConstants = {
    TRANSACTIONS_REQUEST: 'TRANSACTIONS_REQUEST',
    TRANSACTIONS_SUCCESS: 'TRANSACTIONS_SUCCESS',
    TRANSACTIONS_FAILURE: 'TRANSACTIONS_FAILURE',
    TOTAL_TRANSACTIONS_SUCCESS: 'TOTAL_TRANSACTIONS_SUCCESS',
}

export const cashbackConstants = {
    CASHBACK_REQUEST: 'CASHBACK_REQUEST',
    CASHBACK_SUCCESS: 'CASHBACK_SUCCESS',
    CASHBACK_FAILURE: 'CASHBACK_FAILURE',
    TOTAL_CASHBACKS_SUCCESS: 'TOTAL_CASHBACKS_SUCCESS'
}

export const getBalanceConstants = {
    GET_BALANCE_SUCCESS: 'GET_BALANCE_SUCCESS'
}