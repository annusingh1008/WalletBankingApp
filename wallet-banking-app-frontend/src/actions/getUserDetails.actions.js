// import { getUserDetailsConstants } from "./constants";
// import axios from "axios";

// export const getUserDetails = (email) => {
//   return async (dispatch) => {
//     const res = await axios.get(
//       `http://localhost:8085/getUserDetails/${email}`
//     );

//     if (res.status === 200) {
//       dispatch({
//         type: getUserDetailsConstants.GET_USER_DETAILS_SUCCESS,
//         payload: res.data,
//       });
//     }
//   };
// };
