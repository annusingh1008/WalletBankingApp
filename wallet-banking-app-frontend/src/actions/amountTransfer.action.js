import axios from "axios";
import { amountTransferConstants } from "./constants";

export const amountTransfer = (amountDetails) => {
  return async (dispatch) => {
    const res = await axios.post("http://localhost:8085/transfer/amount", {
      ...amountDetails,
    });

    if (res.status === 200) {
      if (res.data === "User does not exist") {
        alert("User does not exist! \nPlease check your email.");
      } else if (res.data === "Amount Transferred Successfully...!!") {
        alert("Transaction Successful");

        dispatch({
          type: amountTransferConstants.AMOUNT_TRANSFER_SUCCESS,
          payload: {
            amount:
              parseInt(amountDetails.currentAmount) -
              parseInt(amountDetails.amount),
            email: amountDetails.email,
          },
        });
      }
    }
  };
};
