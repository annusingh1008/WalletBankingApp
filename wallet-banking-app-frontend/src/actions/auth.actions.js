import axios from "axios";
import { authConstants } from "./constants";

export const login = (user) => {
  return async (dispatch) => {
    const res = await axios.post("http://localhost:8085/signin", {
      ...user,
    });

    if (res.status === 200) {
      if (res.data === "Signed in Successfully...!!") {
        localStorage.setItem("user", JSON.stringify(user));
        localStorage.setItem("email", user.email);

        dispatch({
          type: authConstants.LOGIN_SUCCESS,
          payload: user,
        });
      } else if (res.data === "Account does not exists with this email...!!") {
        alert("Account does not exists!! \nPlease check your email..");
      } else if (res.data === "Incorrect Password...!!") {
        alert("Please check your Password!!");
      }
    }
  };
};

export const getAuthDetails = (state) => state.auth;

export const signout = () => {
  return async (dispatch) => {
    dispatch({ type: authConstants.LOGOUT_REQUEST });
    localStorage.clear();
    dispatch({ type: authConstants.LOGOUT_SUCCESS });
  };
};
