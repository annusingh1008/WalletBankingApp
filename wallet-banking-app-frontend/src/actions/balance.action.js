import axios from "axios";
import { getBalanceConstants } from "./constants";

export const getUserBalance = (email) => {
  return async (dispatch) => {
    const res = await axios.get(`http://localhost:8085/getBalance/${email}`);

    dispatch({
      type: getBalanceConstants.GET_BALANCE_SUCCESS,
      payload: res.data,
    });
  };
};
