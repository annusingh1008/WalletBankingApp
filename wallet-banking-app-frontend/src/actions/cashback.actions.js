import axios from "axios";
import { cashbackConstants } from "./constants";

export const getAllCashBacks = (currentPage, email) => {
  return async (dispatch) => {
    const userEmail = localStorage.getItem("email");
    const res = await axios.get(
      `http://localhost:8085/getAllCashbacks/${userEmail}?pageNumber=${currentPage}&pageSize=${10}`
    );

    dispatch({
      type: cashbackConstants.CASHBACK_SUCCESS,
      payload: res.data,
    });
  };
};

export const getCashbacks = (state) => state.cashbacks;

export const getTotalCashbacks = (email) => {
  return async (dispatch) => {
    const res = await axios.get(
      `http://localhost:8085/getTotalCashbacks/${email}`
    );

    dispatch({
      type: cashbackConstants.TOTAL_CASHBACKS_SUCCESS,
      payload: res.data,
    });
  };
};
