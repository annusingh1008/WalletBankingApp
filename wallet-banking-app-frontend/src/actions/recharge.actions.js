import axios from "axios";
import { rechargeConstants } from "./constants";

export const recharge = (rechargeDetails) => {
  return async (dispatch) => {
    const res = await axios.post("http://localhost:8085/recharge", {
      ...rechargeDetails,
    });

    if (res.status === 200) {
      if (res.data === "Amount Credited Successfully...!!") {
        alert("Recharged Successful");

        dispatch({
          type: rechargeConstants.RECHARGE_SUCCESS,
          payload: {
            email: rechargeDetails.email,
            amount:
              parseInt(rechargeDetails.amount) +
              parseInt(rechargeDetails.currentAmount),
          },
        });
      }
    }
  };
};
