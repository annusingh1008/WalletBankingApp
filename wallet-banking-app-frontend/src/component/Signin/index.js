import React, { useState } from "react";
import { Container, Row, Col, Form, Button } from "react-bootstrap";
import Layout from "../Layout";
import { getAuthDetails, login } from "../../actions/auth.actions";
import { useDispatch, useSelector } from "react-redux";
import "./style.css";
import { Redirect } from "react-router-dom";

const Signin = () => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  const dispatch = useDispatch();
  const auth = useSelector(getAuthDetails);

  const userLogin = (e) => {
    e.preventDefault();
    const user = { email, password };

    dispatch(login(user));
  };

  if (auth.authenticate) {
    return <Redirect to={"/"} />;
  }

  return (
    <Layout>
      <Container style={{ marginTop: "110px" }}>
        <Row style={{ marginLeft: "18%" }}>
          <Col>
            <Form data-testid="input-form" onSubmit={userLogin}>
              <label>Email</label>
              <br />
              <input
                data-testid="inputEmail"
                className="input"
                placeholder="Email"
                value={email}
                type="email"
                onChange={(e) => setEmail(e.target.value)}
                required
              />
              <br />
              <label className="label">Password</label>
              <br />
              <input
                data-testid="inputPassword"
                className="input"
                placeholder="Password"
                value={password}
                type="password"
                onChange={(e) => setPassword(e.target.value)}
                required
              />{" "}
              <br />
              <Button
                data-testid="submit-btn"
                className="btn"
                variant="primary"
                type="submit"
              >
                Submit
              </Button>
            </Form>
          </Col>
        </Row>
      </Container>
    </Layout>
  );
};

export default Signin;
