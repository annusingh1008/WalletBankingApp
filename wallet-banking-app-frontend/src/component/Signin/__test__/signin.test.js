import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import Signin from "../index";
import { Provider } from "react-redux";
import store from "../../../store/index";
import { BrowserRouter as Router } from "react-router-dom";
import { getAuthDetails, login } from "../../../actions/auth.actions";

const mockDispatch = jest.fn();
jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: () => mockDispatch,
}));

jest.mock("../../../actions/auth.actions", () => ({
  getAuthDetails: jest.fn(() => ({
    user: {
      firstName: "Annu",
      lastName: "Singh",
      email: "annu@gmail.com",
      password: "12345678",
      amount: "1000",
    },
    authenticate: false,
    loading: false,
    error: null,
  })),
  login: jest.fn((payload) => ({
    type: "login",
    payload,
  })),
}));

describe("Signin", () => {
  it("should render email and password textarea element", () => {
    render(
      <Provider store={store}>
        <Router>
          <Signin />
        </Router>
      </Provider>
    );

    const textAreaElement1 = screen.getByPlaceholderText(/Email/i);
    expect(textAreaElement1).toBeInTheDocument();
    const textAreaElement2 = screen.getByPlaceholderText(/Password/i);
    expect(textAreaElement2).toBeInTheDocument();
  });

  it("should render email and password label element", () => {
    render(
      <Provider store={store}>
        <Router>
          <Signin />
        </Router>
      </Provider>
    );

    const labeElement1 = screen.getByText("Email");
    expect(labeElement1).toBeInTheDocument();
    const labeElement2 = screen.getByText("Password");
    expect(labeElement2).toBeInTheDocument();
  });

  it("should be able to type email and password", () => {
    render(
      <Provider store={store}>
        <Router>
          <Signin />
        </Router>
      </Provider>
    );

    const textAreaElement1 = screen.getByPlaceholderText(/Email/i);
    fireEvent.change(textAreaElement1, { target: { value: "annu@gmail.com" } });
    expect(textAreaElement1.value).toBe("annu@gmail.com");

    const textAreaElement2 = screen.getByPlaceholderText(/Password/i);
    fireEvent.change(textAreaElement2, { target: { value: "1234" } });
    expect(textAreaElement2.value).toBe("1234");
  });

  it("should render submit button", () => {
    render(
      <Provider store={store}>
        <Router>
          <Signin />
        </Router>
      </Provider>
    );

    const buttonElement = screen.getByText("Submit");
    expect(buttonElement).toBeInTheDocument();
  });

  it("renders properly", () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <Signin />
        </Router>
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("Form can be submitted", () => {
    const { queryByTestId } = render(
      <Provider store={store}>
        <Router>
          <Signin />
        </Router>
      </Provider>
    );
    fireEvent.click(queryByTestId("submit-btn"));
    fireEvent.submit(queryByTestId("input-form"));
    expect(mockDispatch).toHaveBeenCalled();
  });

  it("authenticate is true ", () => {
    getAuthDetails.mockImplementationOnce(() => ({
      user: {
        firstName: "Annu",
        lastName: "Singh",
        email: "annu@gmail.com",
        password: "12345678",
        amount: "1000",
      },
      authenticate: true,
      loading: false,
      error: null,
    }));
    const { container } = render(
      <Provider store={store}>
        <Router>
          <Signin />
        </Router>
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
