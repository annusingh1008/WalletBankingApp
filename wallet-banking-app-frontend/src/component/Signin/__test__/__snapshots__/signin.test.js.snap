// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signin authenticate is true  1`] = `null`;

exports[`Signin renders properly 1`] = `
<nav
  class="navbar navbar-expand-lg navbar-dark bg-dark fixed-top"
  style="z-index: 2;"
>
  <div
    class="container-fluid"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      Wallet Banking
    </a>
  </div>
  <button
    aria-controls="responsive-navbar-nav"
    aria-label="Toggle navigation"
    class="navbar-toggler collapsed"
    type="button"
  >
    <span
      class="navbar-toggler-icon"
    />
  </button>
  <div
    class="navbar-collapse collapse"
    id="responsive-navbar-nav"
  >
    <div
      class="navbar-nav"
    >
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          href="/signin"
        >
          Signin
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          href="/signup"
        >
          Signup
        </a>
      </li>
    </div>
  </div>
</nav>
`;
